src/ttsim/plot/dag/tt.py

```diff
@@ -229,49 +275,53 @@ def _get_node_descriptions(
+            descr = textwrap.fill(descr, width=79)
+            out[p] = descr
     return out
 
 
-def _create_dag_with_selected_nodes(
-    complete_dag: nx.DiGraph,
-    node_selector: _QNameNodeSelector,
+def select_nodes_from_dag(
+    complete_tt_dag: nx.DiGraph,
+    tt_targets_qnames: set[str],
+    selection_type: Literal["neighbors", "descendants", "ancestors", "nodes"],
+    selection_depth: int | None = None,
 ) -> nx.DiGraph:
     """Select nodes based on the node selector."""
     selected_nodes: set[str] = set()
-    if node_selector.type == "nodes":
-        selected_nodes.update(node_selector.qnames)
-    elif node_selector.type == "ancestors":
-        for node in node_selector.qnames:
+    if selection_type == "nodes":
+        selected_nodes.update(tt_targets_qnames)
+    elif selection_type == "ancestors":
+        for node in tt_targets_qnames:
+            selected_nodes.update(
+                _kth_order_predecessors(complete_tt_dag, node, order=selection_depth)
+                if selection_depth
+                else [*list(nx.ancestors(complete_tt_dag, node)), node]
+            )
+    elif selection_type == "descendants":
+        for node in tt_targets_qnames:
             selected_nodes.update(
-                _kth_order_predecessors(complete_dag, node, order=node_selector.order)
-                if node_selector.order
-                else [*list(nx.ancestors(complete_dag, node)), node]
+                _kth_order_successors(complete_tt_dag, node, order=selection_depth)
+                if selection_depth
+                else [*list(nx.descendants(complete_tt_dag, node)), node]
             )
-    elif node_selector.type == "descendants":
-        for node in node_selector.qnames:
+    elif selection_type == "neighbors":
+        order = selection_depth or 1
+        for node in tt_targets_qnames:
             selected_nodes.update(
-                _kth_order_successors(complete_dag, node, order=node_selector.order)
-                if node_selector.order
-                else [*list(nx.descendants(complete_dag, node)), node]
+                _kth_order_neighbors(complete_tt_dag, node, order=order)
             )
-    elif node_selector.type == "neighbors":
-        order = node_selector.order or 1
-        for node in node_selector.qnames:
-            selected_nodes.update(_kth_order_neighbors(complete_dag, node, order=order))
     else:
         msg = (
-            f"Invalid node selector type: {node_selector.type}. "
+            f"Invalid selection type: {selection_type}. "
             "Choose one of 'nodes', 'ancestors', 'descendants', or 'neighbors'."
         )
         raise ValueError(msg)
 
-    dag_copy = copy.deepcopy(complete_dag)
-    dag_copy.remove_nodes_from(set(complete_dag.nodes) - set(selected_nodes))
+    dag_copy = copy.deepcopy(complete_tt_dag)
+    dag_copy.remove_nodes_from(set(complete_tt_dag.nodes) - set(selected_nodes))
     return dag_copy
```

Comment:
```suggestion
    return copy.deepcopy(complete_tt_dag).remove_nodes_from(set(complete_tt_dag.nodes) - set(selected_nodes))
```
If you find that to be too much for one line, create:

```py
unnecessary_nodes = set(complete_tt_dag.nodes) - set(selected_nodes)
```

upfront. Makes the logic clear instead of implementation details.

Reply:
`remove_nodes_from` is an inplace method which is why I do it this way. Returning and
removing nodes in the same line won't work I believe

Reply:
My bad, sorry for the noise!